File: multi-gammon-core/src/test/java/org/akoshterek/backgammon/agent/gnubg/TestGnuBg.java
package org.akoshterek.backgammon.agent.gnubg;

import org.akoshterek.backgammon.agent.Agent;
import org.akoshterek.backgammon.board.Board;
import org.akoshterek.backgammon.board.PositionClass;
import org.akoshterek.backgammon.eval.Evaluator;
import org.akoshterek.backgammon.eval.Reward;
import org.junit.Assert;
import org.junit.BeforeClass;
import org.junit.Test;

import java.nio.file.Path;
import java.nio.file.Paths;

/**
 * @author Alex
 *         date 18.09.2015.
 */
public class TestGnuBg {
    @Test
    public void testRace() {
        Agent agent = new GnubgAgent(Evaluator.getInstance().getBasePath());
        String positionId = "4PPBQRCw58gBMA";
        System.out.println(positionId);
        Board board = Board.positionFromID(positionId);
        PositionClass pc = Evaluator.getInstance().classifyPosition(board);
        Assert.assertTrue(pc == PositionClass.CLASS_CONTACT);
        Reward reward = agent.evaluatePosition(board, pc);
        System.out.println("Reward: " + reward.toString());
    }

    @BeforeClass
    public static void init() {
        Path currentPath = Paths.get("").toAbsolutePath().normalize();
        Evaluator.getInstance().setSeed(16000000L);
        Evaluator.getInstance().load(currentPath);
    }
}

File: multi-gammon-core/src/test/java/org/akoshterek/backgammon/agent/gnubg/TestGnuBg.scala
package org.akoshterek.backgammon.agent.gnubg

import java.nio.file.Paths
import org.akoshterek.backgammon.board.{Board, PositionClass}
import org.akoshterek.backgammon.dice.PseudoRandomDiceRoller
import org.akoshterek.backgammon.eval.Evaluator
import org.junit.{Assert, BeforeClass, Test}

/**
  * @author Alex
  *         date 18.09.2015.
  */
object TestGnuBg {
  @BeforeClass def init(): Unit = {
    val currentPath = Paths.get("").toAbsolutePath.normalize
    Evaluator.diceRoller = PseudoRandomDiceRoller()
    Evaluator.basePath = currentPath
  }
}

class TestGnuBg {
  @Test def testRace(): Unit = {
    val agent = new GnubgAgent(Evaluator.basePath)
    val positionId = "4PPBQRCw58gBMA"
    val board = Board.positionFromID(positionId)
    val pc = Evaluator.classifyPosition(board)
    Assert.assertTrue(pc == PositionClass.CLASS_CONTACT)
    agent.evaluatePosition(board, pc)
  }
}